From 6a5887919f00da84c973ec61c59efcd7d0fb120e Mon Sep 17 00:00:00 2001
From: Gopi Kumar Bulusu <gopi@sankhya.com>
Date: Mon, 1 Apr 2024 16:21:28 +0530
Subject: [PATCH 48/54] Fix build issues after Xilinx 2023.2 binutils patch
 merge

binutils/readelf.c - duplicate case statement
gas/config/tc-microblaze.c - Missing , between array elements
gas/config/tc-microblaze.c - A whole hunk ended up in wrong function/switch

Signed-off-by: Aayush Misra <aayushm@amd.com>
---
 bfd/libbfd.h               |   6 +-
 binutils/readelf.c         |   5 -
 gas/config/tc-microblaze.c | 375 +++++++++++++++++++------------------
 3 files changed, 192 insertions(+), 194 deletions(-)

Index: gdb-14.2/bfd/libbfd.h
===================================================================
--- gdb-14.2.orig/bfd/libbfd.h
+++ gdb-14.2/bfd/libbfd.h
@@ -3005,6 +3005,9 @@ static const char *const bfd_reloc_code_
   "BFD_RELOC_MACH_O_ARM64_GOT_LOAD_PAGE21",
   "BFD_RELOC_MACH_O_ARM64_GOT_LOAD_PAGEOFF12",
   "BFD_RELOC_MACH_O_ARM64_POINTER_TO_GOT",
+  "BFD_RELOC_MICROBLAZE_64",
+  "BFD_RELOC_MICROBLAZE_EA64",
+  "BFD_RELOC_MICROBLAZE_64_PCREL",
   "BFD_RELOC_MICROBLAZE_32_LO",
   "BFD_RELOC_MICROBLAZE_32_LO_PCREL",
   "BFD_RELOC_MICROBLAZE_32_ROSDA",
@@ -3013,13 +3016,12 @@ static const char *const bfd_reloc_code_
   "BFD_RELOC_MICROBLAZE_32_NONE",
   "BFD_RELOC_MICROBLAZE_64_NONE",
   "BFD_RELOC_MICROBLAZE_64_GOTPC",
+  "BFD_RELOC_MICROBLAZE_64_GPC",
   "BFD_RELOC_MICROBLAZE_64_GOT",
   "BFD_RELOC_MICROBLAZE_64_PLT",
   "BFD_RELOC_MICROBLAZE_64_GOTOFF",
   "BFD_RELOC_MICROBLAZE_32_GOTOFF",
   "BFD_RELOC_MICROBLAZE_COPY",
-  "BFD_RELOC_MICROBLAZE_64",
-  "BFD_RELOC_MICROBLAZE_64_PCREL",
   "BFD_RELOC_MICROBLAZE_64_TLS",
   "BFD_RELOC_MICROBLAZE_64_TLSGD",
   "BFD_RELOC_MICROBLAZE_64_TLSLD",
